### DISCUSSION TITLE
Determine if Bluetooth Exists on Device

### DISCUSSION THREAD

**Discussion by woodrufs** [kenjdavidson/react-native-bluetooth-classic#106] [Ideas]
Hi Ken,
Thank you for your hard work on this project! I have been attempting to integrate it into an app I am migrating from RN .59 to .63, and the process has been smooth sailing so far using this (I replaced an in-house, half baked solution with this library). Thanks for that 😃 
I do have one question, though. I am hoping to determine if the device I am using has a bluetooth status of either: Enabled, Disabled, or 'device doesn't have bluetooth'. Is there a good solution that I am not seeing in order to determine when a device doesn't have bluetooth? I have some zebra scanners that don't use bluetooth and instead have their own SDK to intercept a scanned barcode. I am hoping to hide my bluetooth menu in a scenario like this.
Any input is really appreciated. Thanks again!

**Comment by kenjdavidson** [comment]
No, sadly there is nothing in there currently to determine if the device actually has bluetooth or not, that's kind of lumped into the `isBluetoothEnabled()` where:
- mAdapter == null (no bluetooth adapter)
- `mAdatper.isEnabled()` turned off

So at this point no.  With that said, I see no reason why there can't be another request:

`isBluetoothAvailable()` 

that would return true/false based on whether it's available.  Or would it be better to FAIL the `isBluetoothEnabled` with an exception if the device doesn't have bluetooth (a breaking change in most cases) but probably a better bet:

`isBluetoothEnabled()`
- True yes
- False no
- Exception no bluetooth

People should already be wrapping `isBluetoothEnabled()` in a try/catch anyhow, since it's a promise, so it shouldn't mess too many people up.  I think the IOS version is already throwing an error if there is no bluetooth.

Feel free to throw this into a pull request if you end up working on it.  I'll have to look into seeing if this is available (or even meaningful) on IOS devices.

Another option is, getting lucky with the:

`requestBluetoothEnabled()` which may fail in a way that provides details about Bluetooth not being available.

**Reply by woodrufs** [reply]
I gave an attempt at this. My PR is #107. I do not think this is relevant on iOS because I think all iOS devices have Bluetooth.
